frontend/src/features/documents/DownloadDocumentButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { LinkButton } from 'components/common/buttons';
+import LoadingBackdrop from 'components/maps/leaflet/LoadingBackdrop/LoadingBackdrop';
+import { FileDownload } from 'models/api/DocumentManagement';
+import { FaDownload } from 'react-icons/fa';
+
+import { useDocumentProvider } from './documentDetail/hooks/useDocumentProvider';
+
+export interface IDownloadDocumentButtonProps {
+  mayanDocumentId: number;
+  mayanFileId?: number;
+}
+
+const DownloadDocumentButton: React.FunctionComponent<IDownloadDocumentButtonProps> = props => {
+  let provider = useDocumentProvider();
+
+  async function DownloadFile(mayanDocumentId: number, mayanFileId?: number) {
+    if (mayanFileId !== undefined) {
+      const data = await provider.downloadDocumentFile(mayanDocumentId, mayanFileId);
+      if (data) {
+        showFIle(data.payload);
+      }
+    } else {
+      const data = await provider.downloadDocumentFileLatest(mayanDocumentId);
+      if (data) {
+        showFIle(data.payload);
+      }
+    }
+  }
+
+  const showFIle = (file: FileDownload) => {
+    const aElement = document.createElement('a');
```

Comment:
@FuriousLlama have you tested this in FF? If I remember correctly you may need to attach the a to the dom for this to work in FF.

Reply:
I dont have FF atm. But will give it a try once it installs

Reply:
@devinleighsmith  The download worked on FF